lib/Infer/AliveDriver.cpp

```diff
@@ -156,6 +157,60 @@ struct ReturnLHSRAII {
 };
 }
 
+std::map<souper::Inst *, llvm::APInt>
+performCegisFirstQuery(tools::Transform &t,
+                  std::map<std::string, souper::Inst *> &SouperConsts,
+                  smt::expr &TriedExpr) {
+  IR::Value::reset_gbl_id();
+  IR::State SrcState(t.src);
+  IR::State TgtState(t.tgt);
+  util::sym_exec(SrcState);
+  util::sym_exec(TgtState);
+
+  auto &Sv = SrcState.returnVal();
+  auto &Tv = TgtState.returnVal();
+
+  std::map<souper::Inst *, llvm::APInt> SynthesisResult;
+  SynthesisResult.clear();
+
+  std::set<smt::expr> Vars;
+  std::map<std::string, smt::expr> SMTConsts;
+  for (auto &[Var, Val] : TgtState.getValues()) {
+    auto &Name = Var->getName();
+    if (startsWith("%reserved", Name)) {
+      auto App = Val.first.value.isApp();
+      assert(app);
+      SMTConsts[Name] = (Z3_get_app_arg(smt::ctx(), App, 1));
+    }
+  }
+
+  smt::Solver::check({{(Sv.first.value == Tv.first.value) && (TriedExpr),
+          [&](const smt::Result &R) {
+
+          // no more guesses, stop immediately
+          if (R.isUnsat()) {
+            if (DebugLevel > 3)
+              llvm::errs()<<"No more new possible guesses\n";
+            return;
+          } else if (R.isSat()) {
+            auto &&Model = R.getModel();
+            smt::expr TriedAnte(false);
+
+            for (auto &[name, expr] : SMTConsts) {
+              TriedAnte |= (expr != smt::expr::mkUInt(Model.getInt(expr), expr.bits()));
```

Comment:
does this work for values >64 bits?

Reply:
I'll write a test for this, seems it should work for large const.

Reply:
Unfortunately the Alive driver needs to update to support LHS with width > 64, but my part of code should handle value > 64 bits without change.

Reply:
@manasij7479 please do PRs bumping our alive version from time to time